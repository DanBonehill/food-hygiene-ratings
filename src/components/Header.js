import React from "react";
import {Link} from "react-router-dom";

export const Header = () => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Food Hygiene Ratings</h1>
                </Link>
            </div>
        </div>
    </header>
);
