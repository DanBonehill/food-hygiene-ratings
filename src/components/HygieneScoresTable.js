import React from 'react';
import {connect} from 'react-redux';

const HygieneScoresTable = ({scores, totalScoresCount}) => (
    <div>
        <tr>
            <th>Rating</th>
            <th>Percentage</th>
        </tr>
        {
            scores.filter((score) => score.value !== 0).map((score) => (
                <tr>
                    <td>{score.title}</td>
                    <td>{Math.round(score.value / totalScoresCount * 100)}%</td>
                </tr>
            ))
        }
    </div>
);

const mapStateToProps = (state) => ({
    scores: state.hygieneScores.calculatedScores,
    totalScoresCount: state.hygieneScores.allScores.length
});

export default connect(mapStateToProps)(HygieneScoresTable)