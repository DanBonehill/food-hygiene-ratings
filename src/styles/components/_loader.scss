.loader {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: $l-size;
}
