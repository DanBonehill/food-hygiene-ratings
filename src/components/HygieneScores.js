import React from 'react';
import {connect} from 'react-redux';

import HygieneScoresTable from "./HygieneScoresTable";

// Move this somewhere else
const calculateScores = (hygieneScores) => ({
    "pass": hygieneScores.filter((score) => score === 'Pass').length,
    "exempt": hygieneScores.filter((score) => score === 'Exempt').length,
    "improvement": hygieneScores.filter((score) => score === 'Improvement Required').length,
    "five": hygieneScores.filter((score) => score === "5").length,
    "four": hygieneScores.filter((score) => score === "4").length,
    "three": hygieneScores.filter((score) => score === "3").length,
    "two": hygieneScores.filter((score) => score === "2").length,
    "one": hygieneScores.filter((score) => score === "1").length,
    "zero": hygieneScores.filter((score) => score === "0").length
});

const HygieneScores = ({hygieneScores, selectedAuthority, loading}) => {

    // How to better handle this... Redux?
    const scoresArray = [
        {
            title: "Pass",
            value: calculateScores(hygieneScores).pass
        },
        {
            title: "Improvement Required",
            value: calculateScores(hygieneScores).improvement
        },
        {
            title: "Five",
            value: calculateScores(hygieneScores).five
        },
        {
            title: "Four",
            value: calculateScores(hygieneScores).four
        },
        {
            title: "Three",
            value: calculateScores(hygieneScores).three
        },
        {
            title: "Two",
            value: calculateScores(hygieneScores).two
        },
        {
            title: "One",
            value: calculateScores(hygieneScores).one
        },
        {
            title: "Zero",
            value: calculateScores(hygieneScores).zero
        },
        {
            title: "Exempt",
            value: calculateScores(hygieneScores).exempt
        }
    ];

    return (
        <div>
            {!loading &&
                <div>
                    <h1>{selectedAuthority}</h1>
                    <table>
                        {selectedAuthority &&
                        <HygieneScoresTable scores={scoresArray} totalScoresCount={hygieneScores.length}/>
                        }
                    </table>
                </div>
            }
        </div>
    )
};

const mapStateToProps = (state) => ({
    selectedAuthority: state.authorities.selectedAuthority,
    loading: state.authorities.loading,
    hygieneScores: state.hygieneScores
});

export default connect(mapStateToProps)(HygieneScores)