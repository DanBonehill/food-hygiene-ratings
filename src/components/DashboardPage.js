import React from "react";
import axios from 'axios';
import {connect} from 'react-redux';
import Select from 'react-select';

import {addAuthorities} from '../actions/authorities';
import HygieneScores from "./HygieneScores";
import Loader from './Loader';

export class DashboardPage extends React.Component {
    state = {
        options: [],
        hygieneScores: [],
        selectedAuthority: '',
        loading: false
    };
    componentDidMount() {
        axios.get("http://api.ratings.food.gov.uk/authorities/basic/", {
            headers: {'x-api-version': 2}
        }).then((res) => {
            const options = [];

            res.data.authorities.forEach((authority) => {
                options.push({value: authority.LocalAuthorityId, label: authority.Name})
            });

            this.props.addAuthorities(options);
        }).catch((err) => {
            console.log(err)
        })
    }
    handleChange = (selectedAuthority) => {
        const authorityId = selectedAuthority.value;
        this.setState({selectedAuthority: selectedAuthority.label, loading: true});

        axios.get("http://api.ratings.food.gov.uk/establishments", {
            headers: {'x-api-version': 2},
            params: {
                "localAuthorityId": authorityId,
                "pageNumber": 1,
                "pageSize": 10000
            }
        }).then((res) => {
            const hygieneScores = [];

            res.data.establishments.forEach((establishment) => {
                hygieneScores.push(establishment.RatingValue)
            });

            this.setState({hygieneScores, loading: false})
        })
    };
    render() {
        return (
            <div className="content-container">
                <h3>Select an authority</h3>
                <Select
                    options={this.props.authorities}
                    onChange={this.handleChange}
                />
                {this.state.loading &&
                    <Loader/>
                }
                <HygieneScores
                    hygieneScores={this.state.hygieneScores}
                    selectedAuthority={this.state.selectedAuthority}
                    loading={this.state.loading}
                />
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    authorities: state.authorities
});

const mapDispatchToProps = (dispatch) => ({
    addAuthorities: (authorities) => dispatch(addAuthorities(authorities))
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage)