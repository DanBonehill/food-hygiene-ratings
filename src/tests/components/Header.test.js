import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header';

describe('Header snapshot', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<Header startLogout={() => {}}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
