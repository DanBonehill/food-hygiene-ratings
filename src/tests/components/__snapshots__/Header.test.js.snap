// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot should render correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/"
      >
        <h1>
          Food Hygiene Ratings
        </h1>
      </Link>
    </div>
  </div>
</header>
`;
